src/SerialStream.cpp

```diff
@@ -38,40 +38,47 @@ namespace LibSerial
                                const StopBits&      stopBits) : 
         std::iostream(nullptr)
     {
-        this->Open(fileName); // NOLINT (fuchsia-default-arguments)
-        this->SetBaudRate(baudRate);
-        this->SetCharacterSize(characterSize);
-        this->SetFlowControl(flowControlType);
-        this->SetParity(parityType);
-        this->SetStopBits(stopBits);
+        this->Open(fileName) ;// NOLINT (fuchsia-default-arguments)
+        this->SetBaudRate(baudRate) ;
+        this->SetCharacterSize(characterSize) ;
+        this->SetFlowControl(flowControlType) ;
+        this->SetParity(parityType) ;
+        this->SetStopBits(stopBits) ;
+        this->FlushIOBuffers() ;
     }
 
     SerialStream::~SerialStream() 
     {
         // Close the serial stream if it is open.
         if (this->IsOpen())
         {
-            this->Close();
+            this->FlushIOBuffers() ;
+            this->Close() ;
         }
     }
 
     void
     SerialStream::Open(const std::string& fileName,
                        const std::ios_base::openmode& openMode)
-    try {
-        // Create a new SerialStreamBuf if one does not exist. 
-        if (mIOBuffer == nullptr)
+    {
```

Comment:
I see that you are not a fan of function-try-blocks. :) I have no issues putting the try block inside the function body. FYI, the reasons I tend to use function-try-blocks are (1) most importantly, to catch exceptions from constructor initializer lists, (2) they save one level of horizontal indentation and (3) they will not allow anyone to (accidentally) add any statements to the function after the catch(...) block. Exceptions thrown by such statements will not be caught within the function body.

Reply:
Hi Crayzee Wulf, I can put that back the way you had it.  I was just trying to unify conventions.  I will update them throughout the library and push up the work shortly!

Reply:
Mark: No need to undo them. They are perfectly fine as they are and are equivalent to function-try-blocks in this context.

Reply:
Shoot!  Already done and pushed up! :)

Reply:
Even better. You are fast!